framework/src/meshgenerators/AdvancedExtruderGenerator.C

```diff
@@ -119,6 +119,9 @@ AdvancedExtruderGenerator::validParams()
                         0,
                         "Pitch for helicoidal extrusion around an axis going through the origin "
                         "following the direction vector");
+
+  // Declare that this generator has a generateData method
+  MeshGenerator::setHasGenerateData(params);
```

Comment:
why do we set this?
overriding the generateData is already enough to avoid calling the default version that errors

Reply:
This is a @loganharbour question. Looking at the logic in MeshGenerator.C, an MG does not know that `generateData()` function exists unless `setHasGeneateData(params)` has been set ahead of time in `validParams()`. The logic is set such that `generateData` is called only if  `setHasGeneateData(params)` has been defined for a particular MG. Also, `generateData()` will always get called regardless of whether we are in data driven mode or not.

@loganharbour fee; free to chime in on whether `setHasGenerateData(params)` can be removed and we can get away with only setting `generateData()` for a data-driven MG.